Extending gps:info with information on the source

Frank-Wiebeler opened the pull request:
Adding GPS Endpoint names / associations

Frank-Wiebeler committed:
Extending gps:info with information on the source

Frank-Wiebeler committed:
Change Version of DotNet nuget Package

Frank-Wiebeler committed:
Merge branch 'develop' into feature/gps-naming

saschadoemer commented:
Since the .net build is failing, the PR is converted to draft. Please re-active when the checks are passing.

Frank-Wiebeler committed:
-FIX: Update Protobuf Compiler Version for C# to avoid IBufferMessage Issues

Frank-Wiebeler committed:
Merge branch 'feature/gps-naming' of https://github.com/DKE-Data/agrirouter-tmt-protobuf-definitions into feature/gps-naming

saschadoemer commented on README.adoc:
```diff
@@ -29,6 +29,11 @@ Here are some external resources for the development:
 
+== Usage of these protobuf definitions
+
+These protobuf definitions provide a data structure for data that is sent via agrirouter from one endpoint to one or multiple others. 
```
```suggestion
These protobuf definitions provide a data structure for data that is sent via agrirouter from one endpoint to either one or multiple other endpoints. 
```

saschadoemer commented on dotnet/agrirouter/protobuf/technicalmessagetype/gps.proto:
```diff
@@ -18,8 +18,8 @@ message GPSList{
             // 9-13 Reserved
-            D_ERROR = 14;//Error receiving GPS position from the receiver
-            D_NOT_AVAILABLE = 15;
+            D_ERROR = 14; //Error receiving GPS position from the receiver
```
```suggestion
            D_ERROR = 14;//Error receiving GPS position from the receiver
```

To match the other formats I would remove the whitespace.

saschadoemer commented on dotnet/AgrirouterTMTProtobufDefinitions/AgrirouterTMTProtobufDefinitions/AgrirouterTMTProtobufDefinitions.csproj:
```diff
@@ -12,11 +12,11 @@
         <Product>agrirouter-tmt-protobuf-definitions</Product>
         <RepositoryUrl>https://github.com/DKE-Data/agrirouter-tmt-protobuf-definitions</RepositoryUrl>
-        <PackageVersion>2.1.0</PackageVersion>
+        <PackageVersion>2.2.0</PackageVersion>
```
Since the version is changing only within the release process I would recommend not changing the version within the PR.

Frank-Wiebeler replied:
Fixed

saschadoemer commented on java/pom.xml:
```diff
@@ -4,7 +4,7 @@
     <groupId>com.agrirouter.proto</groupId>
     <artifactId>agrirouter-tmt-protobuf-definitions</artifactId>
-    <version>2.1.0-SNAPSHOT</version>
+    <version>2.2.0-SNAPSHOT</version>
```
Since the version is changing only within the release process I would recommend not changing the version within the PR.

Frank-Wiebeler replied:
Fixed

Frank-Wiebeler committed:
Update README.adoc

Co-authored-by: Sascha Doemer <Cf4ThQgxcnxAovOUXTNv@saschadoemer.de>

Frank-Wiebeler committed:
Update dotnet/agrirouter/protobuf/technicalmessagetype/gps.proto

Co-authored-by: Sascha Doemer <Cf4ThQgxcnxAovOUXTNv@saschadoemer.de>

Frank-Wiebeler committed:
Reset version number

Frank-Wiebeler committed:
Merge branch 'feature/gps-naming' of https://github.com/DKE-Data/agrirouter-tmt-protobuf-definitions into feature/gps-naming

saschadoemer commented on README.adoc:
```diff
@@ -29,6 +29,11 @@ Here are some external resources for the development:
 
+== Usage of these protobuf definitions
+
+These protobuf definitions provide a data structure for data that is sent via agrirouter from one endpoint to either one or multiple other endpoints. 
```
> 
```suggestion
These protobuf definitions provide a structure for data that is sent via agrirouter from one endpoint to either one or multiple endpoints. 
```

saschadoemer commented on dotnet/agrirouter/protobuf/technicalmessagetype/gps.proto:
```diff
@@ -36,8 +36,18 @@ message GPSList{
         uint32 number_of_satellites = 7;
         google.protobuf.Timestamp gps_utc_timestamp = 8;
         FieldStatus field_status = 9;//Are we on a road, in a field or just somewhere offroad?
+        uint32 source_device_index = 10;//The index of the source device in the SourceDeviceList
```
```suggestion
        uint32 source_device_index = 10;//The index of the source device in the list of source devices.
```
I think it is better to avoid technical terms in functional descriptions.

saschadoemer commented on java/src/main/resources/com/dke/data/agrirouter/proto/technicalmessagetype/gps.proto:
```diff
@@ -36,8 +36,18 @@ message GPSList{
     }
+
+    message SourceDevice{
+        uint32 index = 1;//The index within the list of SourceDevices; Not necessarilly all values are there
```
```suggestion
        uint32 index = 1;//The index within the list of source devices; Not necessarily all values are there
```
I am not quite sure, what this means "Not necessarily all values are there", perhaps this has to be clearer? Does this mean, that the value is optional?

oliverrahner replied:
Sascha's textual comment is not incorporated. I guess what you want to say is that there may be gaps in those indexes? 
Suggestion:
"Indexes are not guaranteed to be consecutive, or of any specific form."

Frank-Wiebeler replied:
Sorry, I didn't see that, it's fixed now, I adjusted the text

saschadoemer commented on java/src/main/resources/com/dke/data/agrirouter/proto/technicalmessagetype/gps.proto:
```diff
@@ -36,8 +36,18 @@ message GPSList{
+    message SourceDevice{
+        uint32 index = 1;//The index within the list of SourceDevices; Not necessarilly all values are there
+        string display_name = 2;//Proprietary name for displaying (might change during operation)
+        string agrirouter_id = 3;//The endpoint_id as mentioned in the ListEndpoints Result
```
```suggestion
        string agrirouter_id = 3;//The agrirouter specific ID of the endpoint as mentioned in the result of the query for listing endpoints.
```

saschadoemer commented on java/src/main/resources/com/dke/data/agrirouter/proto/technicalmessagetype/gps.proto:
```diff
@@ -36,8 +36,18 @@ message GPSList{
+        uint32 index = 1;//The index within the list of SourceDevices; Not necessarilly all values are there
+        string display_name = 2;//Proprietary name for displaying (might change during operation)
+        string agrirouter_id = 3;//The endpoint_id as mentioned in the ListEndpoints Result
+        string internal_id = 4;//A proprietary but static id for sources outside of agrirouter
```
This could be misleading. This is an ID for sources outside the agrirouter, but is named as `internalId`? Perhaps there is a better name for that?

Frank-Wiebeler replied:
Changed to app_specific_id

saschadoemer reviewed (changes requested):
Please have a look at the comments.

Frank-Wiebeler committed:
Update java/src/main/resources/com/dke/data/agrirouter/proto/technicalmessagetype/gps.proto

Co-authored-by: Sascha Doemer <Cf4ThQgxcnxAovOUXTNv@saschadoemer.de>

Frank-Wiebeler committed:
Update README.adoc

Co-authored-by: Sascha Doemer <Cf4ThQgxcnxAovOUXTNv@saschadoemer.de>

Frank-Wiebeler committed:
Update dotnet/agrirouter/protobuf/technicalmessagetype/gps.proto

Co-authored-by: Sascha Doemer <Cf4ThQgxcnxAovOUXTNv@saschadoemer.de>

Frank-Wiebeler committed:
Update java/src/main/resources/com/dke/data/agrirouter/proto/technicalmessagetype/gps.proto

Co-authored-by: Sascha Doemer <Cf4ThQgxcnxAovOUXTNv@saschadoemer.de>

Frank-Wiebeler committed:
Changed internal_id to app_specific_id

Frank-Wiebeler committed:
Generated sources for new version with app_specific_id

Frank-Wiebeler committed:
+ADD: Link to agrirouter docs
-FIX: More precise definition of index
-FIX: aligned java-proto comments with dotnet proto comments
